src/repo/git.go

```diff
@@ -0,0 +1,68 @@
+package repo
+
+import (
+	"fmt"
+	"github.com/go-git/go-git/v5"
+	"net/url"
+	"os"
+	"strings"
+)
+
+//Clone repo component
+
+func (g *Git) Clone(ctx context.Context, url, branch, id string) (path, string, error) {
+
+	var repoURL string
+	var err error
+
+	fmt.Printf("Enter Your GitHub Repo URL : ")
+	fmt.Scan(&repoURL)
+
+	repoName, err := getRepoId(repoURL)
```

Comment:
I would rather remove this call and use url instead.
currently it creates unnecessary complexity